### ISSUE TITLE
Specify steps to reproduce secnonce in test vector

### ISSUE THREAD

**Issue by mplsgrant** [jonasnick/bips#54]
**Problem**

I cannot figure out how to derive the secnonce found in [sign_verify_vectors](https://github.com/jonasnick/bips/blob/musig2/bip-musig2/sign_verify_vectors.json).

**Need**

I want to [re-create the musig2 test suite](https://github.com/mplsgrant/rust-secp256k1-zkp/blob/51fb87455211123eeb89b61ad68e40f83f46b664/src/zkp/musig.rs#L1724) for the rust implementation of secp256k1-zkp, but that implementation does not allow the creation of secnonces out of bytes. Instead, users must derive their secnonces (ostensibly to prevent nonce reuse).

**Solution**

Providing a public record of the parameters to `nonce_gen_internal` should allow me to recreate the secnonce:

- rand_: bytes, 
- sk: Optional[bytes]
- aggpk: Optional[bytes]
- msg: Optional[bytes]
- extra_in: Optional[bytes]

**Comment by jonasnick** [comment]
The `secnonce` in those test vectors was not created with `nonce_gen_internal` but obtained from a random number generator. Therefore, it's practically impossible for me to give you parameters to `nonce_gen_internal` that would result in the same secnonce.

I think it's actually better that the secnonce is not the output of the nonce generation function because otherwise we would tightly couple the test vectors: any slight change in the nonce generation algorithm would require regenerating all signing test vectors.

By the way I'd be very surprised if there's no way to copy raw bytes into `MuSigSecNonce` in the rust code you link to. By the way, this code is just a set of bindings for [libsecp256k1-zkp](https://github.com/ElementsProject/secp256k1-zkp) whose implementation of MuSig2 did not keep up with the rapid progress in the BIP. As a result, it will not pass the test vectors. So the first step here would be to update the libsecp256k1-zkp implementation after we're done making large changes to the BIP.

**Comment by real-or-random** [comment]
> I think it's actually better that the secnonce is not the output of the nonce generation function because otherwise we would tightly couple the test vectors: any slight change in the nonce generation algorithm would require regenerating all signing test vectors.

This is true but when that leads to people being able to use the vectors, that's a problem indeed. But I'm not sure. Any implementation can have internal tests with the test vectors. It's just a question of whether it's possible to use the test vectors in the public API.

Maybe if we implement https://github.com/jonasnick/bips/issues/49, then coupling is not that big of an issue?

**Comment by mplsgrant** [comment]
Thank you both for the replies. I attempted again and failed to derive a SecNonce value using bytes, and I suspect that secp256k1-zkp may require a `secnonce_parse`-style function exposed on the rust side to get that working. In the interim, I plan to use a bespoke secnonce.

In terms of the development differences between secp256k1-zkp and the rapid updates found here, I've found that many of the tests do in fact pass. However, I am keeping an eye on ElementsProject issue 155 to deal with variable length messages and ElementsProject pull 192 to handle nonce generation.
